network-protection/network-protection-subscription-internal/src/main/java/com/duckduckgo/networkprotection/subscription/NetpSubscriptionManager.kt

```diff
@@ -44,4 +50,8 @@ class RealNetpSubscriptionManager @Inject constructor(
         val entitlements = subscriptions.getEntitlementStatus().firstOrNull()
         return@withContext (entitlements?.contains(NetP) == true)
     }
+
+    override fun hasValidEntitlementFlow(): Flow<Boolean> = subscriptions.getEntitlementStatus().map { it.contains(NetP) }
+
+    override suspend fun isTreated(): Boolean = subscriptions.isEnabled()
```

Comment:
Why did we move the `fun isTreated` from implementation detail of `NetworkProtectionAccessState` to a more public interface that only uses `NetworkProtectionAccessState` anyways?

Reply:
My intention was to use NetpSubscriptionManager.kt a a wrapper for the subscription API in the Netp land. However, i didn’t move other direct usages of subscription.isEnabled() to this in this PR. I’ll revert this part for now.